test/object-store/realm.cpp

```diff
@@ -354,7 +354,7 @@ TEST_CASE("SharedRealm: get_shared_realm()") {
         config.schema = util::none;
         auto old_realm = Realm::get_shared_realm(config);
         TestHelper::begin_read(old_realm, old_version);
-        REQUIRE(old_realm->schema().size() == 1);
+        REQUIRE(old_realm->schema().size() == 2);
```

Comment:
Is this hacky? Realm is open in additive mode though... So basically the whole schema is returned.

Reply:
I don't think this change is correct. I would expect the schema at this old version to include any additional properties found on disk _at that version_, but not anything added by a later version. As Thomas pointed out earlier "we need to actually read the schema from the correct transaction version."